app/gui/src/controller/graph/executed.rs

```diff
@@ -451,6 +451,25 @@ impl Handle {
         }
     }
 
+    /// Remove all the connections from the graph. This is a convenience method that calls
+    /// [`disconnect`] for each connection. If any of the calls fails, the first error is
+    /// propagated, but all the connections are attempted to be disconnected.
+    pub fn disconnect_all(&self, connections: impl Iterator<Item = Connection>) -> FallibleResult {
+        let errors =
+            connections.map(|c| self.disconnect(&c)).filter_map(|r| r.err()).collect::<Vec<_>>();
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            // Failure has no good way to propagate multiple errors with `Failure`. So we propagate
+            // only the first one.
+            let first_error = errors
+                .into_iter()
+                .next()
+                .expect("Errors are not empty, so there must be a first element.");
+            Err(first_error)
+        }
+    }
```

Comment:
why not `if let Some(error) = errors.into_iter().first()`?

Reply:
Much nicer! Unfortunately, `first` only exists on `Vec` not on `IntoIter` so I'll have to keep that. Or do a `take`, which will mutate the `Vec` and thus is not really better.

Reply:
Oh of course, `next`